libsentrykube/kube.py

```diff
@@ -104,26 +107,57 @@ def _consolidate_variables(
     """
     We have multiple levels of overrides for our value files.
     1. The values defined inside the service directory as values.yaml.
-    2. overridden by the regional overrides in
+    2. overridden by creating a hierarchical structure. Adding an intermediate directory
+       in 'region_override' with a '_values.yaml' file allows to have a common config
+       between a set of regions. This is useful if regions in a service are different, but
+       subset of them are similar.
+    3. overridden by a common '_values.yaml' within a region folder that has a shared config
+       between all clusters in the region.
+    4. overridden by the regional overrides in
        `service/region_override/region/cluster.yaml`
-    3. overridden by the managed override file. This is like point 2. Conceptually
+    5. overridden by the managed override file. This is like point 2. Conceptually
        there is no difference, practically this is managed by tools while region
        overrides are managed manually and they can contain comments. Tools cannot
        preserve comments.
-    4. overridden by the cluster file. Which is likely going to be replaced by 2 and 3.
+    6. overridden by the cluster file. Which is likely going to be replaced by 2 and 3.
 
     TODO: write the minimum components of a yaml parser to remove step 3 and
           patch the regional override preserving comments.
     """
 
+    # check that there is a single cluster file per customer
+    assert_single_cluster_for_customer(
+        service_name, customer_name, cluster_name, external
+    )
+
     # Service defaults from _values
     service_values = get_service_values(service_name, external)
 
     # Service data overrides from services/SERVICE/region_overrides/
     service_value_overrides = get_service_value_overrides(
         service_name, customer_name, cluster_name, external
     )
-    deep_merge_dict(service_values, service_value_overrides)
+
+    if service_value_overrides:
```

Comment:
This does not seem right.
This way `services/SERVICE/region_overrides/REGION/_values.yaml` is taken into account only if there is also a cluster specific override file. 

If we go from generic to specific, we should apply `services/SERVICE/region_overrides/REGION/_values.yaml` even if there is no more specific cluster specific region override (`services/SERVICE/region_overrides/REGION/my_cluster.yaml`)

So I think it should be:
```
common_service_values = get_common_regional_override(...)
service_value_overrides = get_service_value_overrides(...)
deep_merge_dict(common_service_values, service_value_overrides)


```

Reply:
I was under the impression that there is always a necessity for a cluster specific override, which I now realise was incorrect.
I will change accordingly